### PR TITLE
#18801 add "count" field to GraphQL

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Collections;
+import java.util.Map;
+
+public enum CountTypeProvider implements GraphQLTypesProvider {
+
+    INSTANCE;
+
+    final Map<String, GraphQLOutputType> pageFields = map(
+            "totalCount", GraphQLString,
+            "fieldName", GraphQLString);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java:20]: I am not sure if this should be fieldName, maybe 'collectionName' could be a better name

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+public enum CountTypeProvider implements GraphQLTypesProvider {
+
+    INSTANCE;
+
+    final Map<String, GraphQLOutputType> pageFields = map(
+            "totalCount", GraphQLString,
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java:19]: should it be the same name than here https://github.com/dotCMS/core/pull/19032/files#diff-cf87643edb0d8a8ccf07993e94b7180bR21?

maybe we can create a suoer class here to define to add the count field and create the Map, and later each sub class could add the extra parameters

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/CountDataFetcher.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import graphql.schema.DataFetchingEnvironment;
+
+public class CountDataFetcher implements DataFetcher<List<Map<String, Object>>> {
+    @Override
+    public List<Map<String, Object>> get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            return ((DotGraphQLContext) environment.getContext()).getFieldCountMaps();
+        } catch (Exception e) {
+            Logger.error(this, e.getMessage(), e);
+            throw e;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/CountDataFetcher.java:24]: maybe should re throw a DotRuntimeException

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java:20]: Waiting on discussion with team

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java:19]: Waiting on discussion with team

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java]: leaving like this

File: dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/business/CountTypeProvider.java]: leaving it like this

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by freddyucv — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/business/QueryMetadataFieldProvider.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;
+import static graphql.schema.GraphQLList.list;
+
+import com.dotcms.graphql.datafetcher.QueryMetadataDataFetcher;
+import com.dotmarketing.exception.DotDataException;
+import graphql.schema.GraphQLFieldDefinition;
+import java.util.Collection;
+import java.util.Collections;
+
+public enum QueryMetadataFieldProvider implements GraphQLFieldsProvider {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/business/QueryMetadataFieldProvider.java:12]: doc

File: dotCMS/src/main/java/com/dotcms/graphql/business/QueryMetadataTypeProvider.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.dotcms.graphql.util.TypeUtil;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+public enum QueryMetadataTypeProvider implements GraphQLTypesProvider {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/business/QueryMetadataTypeProvider.java:14]: doc

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/QueryMetadataDataFetcher.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+
+import java.util.List;
+import java.util.Map;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+public class QueryMetadataDataFetcher implements DataFetcher<List<Map<String, Object>>> {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/QueryMetadataDataFetcher.java:17]: Doc

**Pull request merged**